core/domain/email_services_test.py

```diff
@@ -57,3 +62,192 @@ def test_reply_info_email_objects_are_created_and_queried_correctly(self):
                 'user_unknown', 'invalid_thread_id'))
 
         self.assertEqual(queried_object, None)
+
+    def test_send_mail_raises_exception_for_invalid_permissions(self):
+        """Tests the send_mail exception raised for invalid user permissions."""
+        send_email_exception = (
+            self.assertRaisesRegexp(
+                Exception, 'This app cannot send emails to users.'))
+        mailgun_api = self.swap(feconf, 'MAILGUN_API_KEY', 'api')
```

Comment:
Maybe name these swap_mailgun_api, swap_mailgun_domain, etc. throughout?

Semantically I'm not sure what "with mailgun_api" means.